vm/systemSmartContracts/esdt.go

```diff
@@ -482,16 +511,158 @@ func (e *esdt) togglePause(args *vmcommon.ContractCallInput, builtInFunc string)
 	return vmcommon.Ok
 }
 
-func (e *esdt) configChange(_ *vmcommon.ContractCallInput) vmcommon.ReturnCode {
-	//TODO: implement me
+func (e *esdt) configChange(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	if !bytes.Equal(args.CallerAddr, e.ownerAddress) {
+		e.eei.AddReturnMessage("configChange can be called by whitelisted address only")
+		return vmcommon.UserError
+	}
+	if args.CallValue.Cmp(zero) != 0 {
+		e.eei.AddReturnMessage("callValue must be 0")
+		return vmcommon.UserError
+	}
+	err := e.eei.UseGas(e.gasCost.MetaChainSystemSCsCost.ESDTOperations)
+	if err != nil {
+		e.eei.AddReturnMessage(err.Error())
+		return vmcommon.OutOfGas
+	}
+	if len(args.Arguments) != 4 {
+		e.eei.AddReturnMessage(vm.ErrInvalidNumOfArguments.Error())
+		return vmcommon.UserError
+	}
+
+	newConfig := &ESDTConfig{
+		OwnerAddress:       args.Arguments[0],
```

Comment:
there is an ownerAddress also in the TokenData. Should there be 2 different ownerAddresses?

Reply:
no. but I wanted to keep them both.

Reply:
ok. those are 2 types of owners :) one is the owner of the ESDT metachain system SC.

Reply:
the other owner is for the actual token.